### PR TITLE
[OC-1025] : complete users admin module

### PR DESCRIPTION
**Pull request by bendaoudmba**
add to release notes:

[OC-1025]: new administration module for users, entities and groups management

### CODE REVIEW

**Comment by freddidierRTE** [comment]
- For UsersController.java , use UserServiceImp as for other controller ? 
- Sonar issues : https://sonarcloud.io/project/issues?branch=OC-1025&id=org.lfenergy.operatorfabric%3Aoperatorfabric-core&resolved=false&types=CODE_SMELL 
- First , Previous , Next / Last , Filter by ..  are not traduce
- When you switch to french , traduction works but when you make a reload , traduction is again in english

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

File: ui/main/src/app/modules/admin/services/login-validator.service.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  isLoginAvailable(control: AbstractControl): Promise<ValidationErrors> | null {
+    return new Promise((resolve, reject) => {
+      this.userService.getAll().subscribe((response) => {
+        if (response.filter(user => user.login === control.value).length > 0) {
+          resolve({ shouldbeUnique: true });
+        } else {
+          resolve(null);
+        }
```
[the remark this answers is not in the document]
Comment (bendaoudmba) [reply ui/main/src/app/modules/admin/services/login-validator.service.ts:22]: use of promise.resolve method is necessary

File: ui/main/src/app/modules/admin/admin-rooting.module.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          //   //  }
+        // ]
+    }
+];
+
+@NgModule({
+    imports: [RouterModule.forChild(routes)],
+    exports: [RouterModule]
+})
+export class AdminRoutingModule { }
```
[the remark this answers is not in the document]
Comment (bendaoudmba) [reply ui/main/src/app/modules/admin/admin-rooting.module.ts:44]: for the moment, we have only /admin==>AdminComponent, but the module should evolve to have more rooting rule in the future

File: ui/main/src/app/services/user.service.ts [hunk 3]
```diff
@@ -8,59 +8,87 @@
...
+export class UserService implements CrudService {
+  readonly userUrl: string;
+  private _userWithPerimeters: UserWithPerimeters;
+  private ngUnsubscribe = new Subject<void>();
+
+  /**
+   * @constructor
+   * @param httpClient - Angular build-in
+   */
+  constructor(private httpClient: HttpClient, private groupsService: GroupsService) {
```
[the remark this answers is not in the document]
Comment (bendaoudmba) [reply ui/main/src/app/services/user.service.ts:33]: done

File: ui/main/package.json [hunk 4]
```diff
@@ -38,17 +38,19 @@
...
     "handlebars": "^4.7.6",
     "jwt-decode": "^2.2.0",
-    "lodash": "^4.17.15",
+    "lodash": "^4.17.20",
     "moment": "^2.26.0",
     "moment-timezone": "^0.5.31",
     "ng-event-source": "^1.0.14",
+    "ng2-table": "^1.3.2",
     "ngrx-router": "^2.0.1",
+    "ngx-bootstrap": "^5.6.1",
```
[the remark this answers is not in the document]
Comment (bendaoudmba) [reply ui/main/package.json:59]: ngx-bootstrap is required for pagination use, i uninstall ng-bootstra instead, and apparently, there is no regression

File: ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    protected translate: TranslateService,
+    private confirmationDialogService: ConfirmationDialogService,
+    protected store: Store<AppState>
+  ) {
+
+    super(modalService, dataService, store);
+    this.crudService = crudService;
+    this.config.sorting = { columns: this.columns };
+    this.getLocale().subscribe(locale => {
+      console.log(locale);
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts:36]: remove this log

File: ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+      this.translate.use(locale);
+      this.translate.get(['admin.input.user.login', 'admin.input.user.firstname', 'admin.input.user.lastname',
+        'admin.input.user.groups', 'admin.input.user.entities', 'admin.input.user.edit', 'admin.input.user.delete',
+        'admin.pagination.firstText', 'admin.pagination.lastText', 'admin.pagination.nextText', 'admin.pagination.prevText'
+        , 'admin.input.user.add', 'admin.input.lines', 'admin.input.user.filter' , 'admin.input.user.filterAll'])
+        .subscribe(translations => {
+          console.log(this.translate.currentLang);
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts:43]: remove this log

File: ui/main/src/app/services/user.service.ts [hunk 7]
```diff
@@ -75,12 +99,23 @@ export class UserService implements CrudService {
         (userWithPerimeters) => {
           if (userWithPerimeters) {
             this._userWithPerimeters = userWithPerimeters;
+            console.log(this._userWithPerimeters);
```
Comment (freddidierRTE) [inline ui/main/src/app/services/user.service.ts:102]: Remove this log

File: ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    this.confirmationDialogService.confirm(
+      this.translate.instant('admin.input.user.confirm'),
+      this.translate.instant('admin.input.delete') + ' ' + row.login + '?',
+      'OK',
+      this.translate.instant('admin.input.user.cancel')
+    ).then((confirmed) => {
+      if (confirmed) {
+        this.deleteItem(row);
+      }
+    }).catch(() => console.log('error)'));
```
Comment (freddidierRTE) [inline ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts:128]: add error information in this log

**Review by freddidierRTE — CHANGES_REQUESTED** [review]

**Comment by freddidierRTE** [comment]
Bug : when creating a user with a space at the end , it create its but it is not possible to delete it after 
Solution --> remove space at the end of the  user login when saving 

+ resolve conflict

**Comment by bendaoudmba** [comment]
well done , i will treat those points this afternoon

File: ui/main/src/app/components/navbar/navbar.component.ts [hunk 9]
```diff
@@ -8,151 +8,157 @@
...
+        this.loadNightModeFromLocalStorage();
+      }
     }
 
-    logOut() {
-        this.store.dispatch(new TryToLogOut());
-    }
+    const hiddenMenus = this.configService.getConfigValue('navbar.hidden', []);
+    this.navigationRoutes = navigationRoutes.filter(route => !hiddenMenus.includes(route.path));
+    this.displayAdmin = this.userService.isCurrentUserAdmin() && this.configService.getConfigValue('admin.hidden');
```
[the remark this answers is not in the document]
Comment (bendaoudmba) [reply ui/main/src/app/components/navbar/navbar.component.ts:108]: done

File: ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts (hunk 5 above)
Comment (bendaoudmba) [reply ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts:36]: done

File: ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts (hunk 6 above)
Comment (bendaoudmba) [reply ui/main/src/app/modules/admin/components/ngtable/users/ofuserstable.component.ts:43]: done

File: ui/main/src/app/services/user.service.ts (hunk 7 above)
Comment (bendaoudmba) [reply ui/main/src/app/services/user.service.ts:102]: done

**Pull request merged**